### PR TITLE
added the transfers controller

### PR DESCRIPTION
**Pull request by emmanueltukpe**
This adds the transfers controller

### CODE REVIEW

File: controllers/transfers.controller.js [hunk 1]
```diff
@@ -32,11 +32,43 @@ const debit = async (account_number, amount) => {
   );
 };
 
+/**
+ * Debits an account by an amount
```
Comment (jirevwe) [inline controllers/transfers.controller.js:36]: The doc string here is incorrect

File: controllers/transfers.controller.js [hunk 2]
```diff
@@ -32,11 +32,43 @@ const debit = async (account_number, amount) => {
...
+ *
+ * @param {String} account_number the account number of the account to be debited
+ * @param {Number} amount the amount to be debited
+ */
+const accountCheck = async (account_number, amount) => {
+  const check = await Accounts.findOne({ account_number });
```
Comment (jirevwe) [inline controllers/transfers.controller.js:42]: The variable name needs to be more descriptive

File: controllers/transfers.controller.js [hunk 3]
```diff
@@ -32,11 +32,43 @@ const debit = async (account_number, amount) => {
...
+  } else {
+    debit(account_number, amount);
+  }
+};
+
 const transfer = async (req, res) => {
   // debit the sender
   // credit the recipient
   // create the transaction
   // return a response ot throw and error
```
Comment (jirevwe) [inline controllers/transfers.controller.js:54]: Delete these lines, they are not really useful you can move them to the docs string if you need them

File: app.js [hunk 4]
```diff
@@ -1,7 +1,9 @@
 //@ts-check
```
Comment (jirevwe) [inline app.js:1]: Remove this and all the `// @ts-ignore`

**Review by jirevwe — CHANGES_REQUESTED** [review]

File: controllers/transfers.controller.js [hunk 5]
```diff
@@ -32,11 +32,43 @@ const debit = async (account_number, amount) => {
...
+  res.status(StatusCodes.CREATED).json({ transaction });
+};
+
+const deposit = async (req, res) => {
+  const transaction = await Transaction.create(req.body);
+  credit(req.body.recipient, req.body.amount);
+  res.status(StatusCodes.CREATED).json({ transaction });
+};
+
+const withdrawal = async (req, res) => {
```
Comment (allengblack) [inline controllers/transfers.controller.js:68]: Looks like you're creating a Transaction here AND in the `accountCheck` function. Is this a requirement?

Comment (emmanueltukpe) [reply controllers/transfers.controller.js:68]: The accountCheck function checks if there is money in that account before debiting

Comment (jirevwe) [reply controllers/transfers.controller.js:64]: Yeah, but on line 65 you are calling `await Transaction.create(req.body)`

**Review by jirevwe — CHANGES_REQUESTED** [review]

**Review by jirevwe — APPROVED** [review]

**Pull request merged**